fvm/programs.go

```diff
@@ -12,27 +12,70 @@ type ProgramEntry struct {
 	Program  *interpreter.Program
 }
 
+type ChangedProgram struct {
+	Address common.Address
+	Name    string
+}
+
+type ProgramGetFunc func(location common.Location) *ProgramEntry
+
+func emptyProgramGetFunc(_ common.Location) *ProgramEntry {
+	return nil
+}
+
+// Programs is a cumulative cache-like storage for Programs helping speed up execution of Cadence
+// Programs don't evict elements at will, like a typical cache would, but it does it only
+// during a cleanup method, which must be called only when the Cadence execution has finished.
+// It it also fork-aware, support cheap creation of children capturing local changes.
 type Programs struct {
-	lock     sync.RWMutex
-	programs map[common.LocationID]ProgramEntry
+	lock       sync.RWMutex
+	programs   map[common.LocationID]ProgramEntry
+	parentFunc ProgramGetFunc
```

Comment:
Is there any particular reason this is a function and not just a pointer to the parent?

Reply:
It's just more concise - otherwise we would have to handle parent being `nil` separately